File: demos/routing/lib/libsodium-port/bench/benchAdd.js
var BENCHCOUNT = 100000;

// require c++ binding module
var wrapper = require('../wrapper.js');

wrapper.ready.then(function () {
  // Find and hash some Points
  var point1 = wrapper.hashToPoint('hello');
  var point2 = wrapper.hashToPoint('world!');

  // Run several tests
  for (var t = 0; t < BENCHCOUNT; t++) {
    point1 = wrapper.pointAdd(point1, point2);
  }

  console.log('success add');
});

File: demos/routing/lib/libsodium-port/bench/benchAdd.c
#define BENCHCOUNT 100000

#include <sodium.h>
#include <stdio.h>

void main() {
  if (sodium_init() == -1) {
    return;
  }

  unsigned char hash1[crypto_generichash_BYTES];
  unsigned char hash2[crypto_generichash_BYTES];
  unsigned char point1[crypto_scalarmult_ed25519_BYTES];
  unsigned char point2[crypto_scalarmult_ed25519_BYTES];

  // Hash
  crypto_generichash(hash1, sizeof hash1, "hello", 5, NULL, 0);
  crypto_core_ed25519_from_uniform(point1, hash1);

  crypto_generichash(hash2, sizeof hash2, "world!", 6, NULL, 0);
  crypto_core_ed25519_from_uniform(point2, hash2);

  // Benchmarks
  for(int i = 0; i < BENCHCOUNT; i++) {
    if (crypto_core_ed25519_add(point1, point1, point2) != 0) {
      printf("FAIL ADD\n");
      return;
    }
  }

  printf("success add\n");
}


